crates/goose/src/providers/ollama.rs

```diff
@@ -154,4 +156,77 @@ impl Provider for OllamaProvider {
         super::utils::emit_debug_trace(&self.model, &payload, &response, &usage);
         Ok((message, ProviderUsage::new(model, usage)))
     }
+
+    /// Generate a session name based on the conversation history
+    /// This override filters out reasoning tokens that some Ollama models produce
+    async fn generate_session_name(&self, messages: &[Message]) -> Result<String, ProviderError> {
+        let message = Message::user().with_text(&self.create_session_name_prompt(messages));
+        let result = self
+            .complete(
+                "You are a title generator. Output only the requested title with no additional text, reasoning, or explanations.",
+                &[message],
+                &[],
+            )
+            .await?;
+
+        let mut description = result.0.as_concat_text();
+        description = Self::filter_reasoning_tokens(&description);
+
+        let sanitized_description = if description.chars().count() > 100 {
+            safe_truncate(&description, 100)
+        } else {
+            description
+        };
+
+        // If the description is too long, the model failed to give us a short description, provide a fallback instead
+        if sanitized_description.split_whitespace().count() > 6 {
```

Comment:
this is not going to work for languages that don't use whitespace to separate words. you could use tiktoken but I'd just leave it at the length check above